src/include/kdbchangetracking.h

```diff
@@ -0,0 +1,47 @@
+#ifndef ELEKTRA_KDBCHANGETRACKING_H
+#define ELEKTRA_KDBCHANGETRACKING_H
+
+#include <kdb.h>
+#include <kdbtypes.h>
+
+#ifdef __cplusplus
+namespace ckdb
+{
+extern "C" {
+#endif
+
+typedef struct _ChangeTrackingContext ChangeTrackingContext;
+typedef struct _KeySetDiff KeySetDiff;
+
+const ChangeTrackingContext * elektraChangeTrackingGetContextFromKdb (KDB * kdb);
+
+KeySetDiff * elektraChangeTrackingCalculateFromContext (KeySet * newKeys, const ChangeTrackingContext * context, Key * parentKey);
+KeySetDiff * elektraChangeTrackingCalculateFromKeySets (KeySet * newKeys, KeySet * oldKeys, Key * parentKey);
+
+void elektraChangeTrackingKeySetDiffDel (KeySetDiff * ksd);
+const Key * elektraChangeTrackingKeySetDiffGetParentKey (const KeySetDiff * ksd);
+KeySetDiff * elektraChangeTrackingCreateKeySetDiff (KeySet * addedKeys, KeySet * removedKeys, KeySet * modifiedKey, Key * parentKey);
+uint16_t elektraChangeTrackingKeySetDiffIncRef (KeySetDiff * ksd);
+uint16_t elektraChangeTrackingKeySetDiffDecRef (KeySetDiff * ksd);
+
+KeySet * elektraChangeTrackingGetAddedKeys (const KeySetDiff * ksd);
+KeySet * elektraChangeTrackingGetRemovedKeys (const KeySetDiff * ksd);
+KeySet * elektraChangeTrackingGetModifiedKeys (const KeySetDiff * ksd); // Returns old keys (pre-modification)
+
+bool elektraChangeTrackingValueChanged (const KeySetDiff * ksd, Key * key);
+bool elektraChangeTrackingMetaChanged (const KeySetDiff * ksd, Key * key);
+
+KeySet * elektraChangeTrackingGetAddedMetaKeys (const KeySetDiff * ksd, Key * key);
+KeySet * elektraChangeTrackingGetRemovedMetaKeys (const KeySetDiff * ksd, Key * key);
+KeySet * elektraChangeTrackingGetModifiedMetaKeys (const KeySetDiff * ksd, Key * key); // Returns old meta keys (pre-modification)
```

Comment:
Since none of these actually involve change tracking (no `ChangeTrackingContext`), I'd move them to a separate header (or even library) for diffs.

I'd say we use the prefix `elektraDiff*`, call the struct `ElektraDiff` and if it's a separate library call it `libelektra-diff`.

Reply:
Yes I was thinking about the same thing. However, not sure how much effort it is to put it in its own library.

Reply:
Shouldn't be too much effort, just a bit of CMake. But you also just leave it in here for now. I'll need to move everything around at some point anyway to make things fit with the new [header concept](https://github.com/ElektraInitiative/libelektra/blob/master/doc/decisions/4_decided/header_file_structure.md)

Reply:
I split it out into separate files. I'll let you do the library then 😅